java-squid/src/main/java/org/sonar/java/DefaultJavaResourceLocator.java

```diff
@@ -132,9 +133,9 @@ public void scanFile(JavaFileScannerContext context) {
         sourceFileByClass.put(classIOFileEntry.getKey(), context.getFileKey());
       }
     }
-    context.addNoSonarLines(javaFilesCache.ignoredLines());
-    ignoredLinesForRules.put(context.getFileKey(), javaFilesCache.ignoredLinesForRules());
     methodStartLines.putAll(javaFilesCache.getMethodStartLines());
+    if (sensorContext != null && javaFilesCache.hasSuppressWarningLines()) {
```

Comment:
I would even go further and add a Precondition.checkNotNull on sensorContext to prevent any possible run without setting it.